pkg/server/router/model.go

```diff
@@ -150,6 +150,13 @@ func (p *Payload) HasMasterKey() bool {
 	return p.AccessKey == MasterAccessKey
 }
 
+// IsValidPluginRequest returns true when the payload indicate that the
+// request is from plugin AND the master key is used
+func (p Payload) IsValidPluginRequest() bool {
+	fromPlugin, _ := p.Data["_from_plugin"].(bool)
```

Comment:
It is the job of a preprocessor to decide whether the request is a valid plugin request, not the `Request` struct. Move this function out of here.